### PR TITLE
add distribution script for rewards

### PR DESCRIPTION
**Pull request by perionjames**
This PR:

1. Adds a distribution script for rewards (that can be sent from a hot wallet)
2. Adds an approval script which must be run once so that the hot wallet can distribute funds via the `BasePool` contract

If we end up using heroku to run this script, we will need to:

1. Top up the wallet with sufficient funds (in both PERC and ETH)
2. Store the private key in heroku's environment config vars (with a strong password and 2FA)
3. Run the approval script once with `$ npm run approve`
4. Setup the heroku scheduler to run `$ npm run distribute` each day at a particular UTC time
5. Periodically topup the hot wallet with funds 

Since the contracts have not yet been deployed, #32 must still be done.

### CODE REVIEW

File: scripts/rewards/Approve.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// TODO fill vars
+import { ethers } from 'hardhat'
+const PERC = "0x60be1e1fe41c1370adaf5d8e66f07cf1c2df2268";
+const sPERC = "";
+const sLPPERC = "";
```
Comment (geoshak) [inline scripts/rewards/Approve.ts:5]: For the sake of docs, and to facilitate the reading, can we add a brief comment of what the variables represent?

I looked for it on the README but couldn't find any refs there either.

File: scripts/rewards/DistributeRewards.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// TODO fill vars
+import { ethers } from 'hardhat'
+import { parseEther } from "ethers/lib/utils";
+const sPERC = "";
+const sLPPERC = "";
+const dailySPERCAllocation = parseEther("1357.4");
+const dailyLPAllocation = parseEther("5429.6");
```
Comment (geoshak) [inline scripts/rewards/DistributeRewards.ts:6]: Just curious, these are arbitrary values, right?
Or is it like a max amount that can be distributed on a single day?

**Review by geoshak — APPROVED** [review]

**Comment by geoshak** [comment]
Ah, one alternative to `heroku` that I learned about recently is [fly.io](fly.io) (a couple of articles [1](https://blog.back4app.com/heroku-vs-fly-io/), [2](https://febriliankr.medium.com/benchmarking-fly-io-vs-render-vs-heroku-ccbdb63b488a)).

Yet another option would be using GCP's Cloud Functions.
It seems to be cost effective:

Selection_099

File: scripts/rewards/DistributeRewards.ts (hunk 2 above)
Comment (perionjames) [reply scripts/rewards/DistributeRewards.ts:6]: No, they are the actual daily drop amounts in PERC for each staking pool.

**Comment by perionjames** [comment]
> Ah, one alternative to `heroku` that I learned about recently is [fly.io](fly.io) (a couple of articles [1](https://blog.back4app.com/heroku-vs-fly-io/), [2](https://febriliankr.medium.com/benchmarking-fly-io-vs-render-vs-heroku-ccbdb63b488a)).
> 
> Yet another option would be using GCP's Cloud Functions. It seems to be cost effective:
> 
> Selection_099

That looks cool! The reason I want to go the heroku route is because I have used it already many times and it will be very easy for me to setup. Since this service will be very simple, the cost of running it is very insignificant (even on heroku).

File: scripts/rewards/Approve.ts (hunk 1 above)
Comment (perionjames) [reply scripts/rewards/Approve.ts:5]: Sure could, these are the two deployed pool contacts.

Comment (geoshak) [reply scripts/rewards/Approve.ts:5]: :pray:

File: scripts/rewards/DistributeRewards.ts (hunk 2 above)
Comment (geoshak) [reply scripts/rewards/DistributeRewards.ts:6]: Got it.

Actually, I don't haha :sweat_smile: 

My understanding of the contract is very lacking.
I'll take some moments to study it's mechanics a little more over the next days.

Thanks man!

**Pull request merged**